### PR TITLE
Doxygen fixes

### PR DESCRIPTION
**Pull request by ValZapod**
See issue #3071

### CODE REVIEW

**Comment by ValZapod** [comment]
@michaelld You can merge, BTW. Also see (builds not under pull reqests but under main build) https://ci.gnuradio.org/buildbot/#/builders/57/builds/144
Dunno how to fix them... It will not be easy, as I am not familiar with those documentation changes. But if you do not want to restore those than we can just delete all \ref.

**Comment by michaelld** [comment]
> @michaelld You can merge, BTW.

Yes, I know as this is documentation! I haven't had time to review!

> Also see (builds not under pull reqests but under main build) https://ci.gnuradio.org/buildbot/#/builders/57/builds/144

I restarted that CI. Not sure what's going on.

> Dunno how to fix them... It will not be easy, as I am not familiar with those documentation changes. But if you do not want to restore those than we can just delete all \ref.

One step at a time!

**Comment by michaelld** [comment]
That CI seems to be going now. Guessing it was some odd network issue that happens from time to time.

**Comment by ValZapod** [comment]
@michaelld No, it is not. I mean these https://github.com/gnuradio/gnuradio/commits/master see red thing, it is because it treats warning like errors... So we are fixing it (and already fixed some of them with prev. doxy commits, compare before and after the fix).

**Comment by michaelld** [comment]
Does this PR fix those red "X" issues with building docs? all of the recent commits on master have this issue, as you note.

**Comment by michaelld** [comment]
Answering my own query: looks like this PR does indeed fix the red "X" issues.

**Comment by michaelld** [comment]
Ok so the doxygen warnings are now (for example):
```
warning: ignoring unsupported tag `OUTPUT_TEXT_DIRECTION  =' at line 102, file /cache/workdir/Debian_10_64_0/build_Debian_10_64/build/gr-audio/swig/audio_swig_doc_swig_docs/Doxyfile
warning: ignoring unsupported tag `JAVADOC_BANNER         =' at line 198, file /cache/workdir/Debian_10_64_0/build_Debian_10_64/build/gr-audio/swig/audio_swig_doc_swig_docs/Doxyfile
warning: ignoring unsupported tag `OPTIMIZE_OUTPUT_SLICE  =' at line 296, file /cache/workdir/Debian_10_64_0/build_Debian_10_64/build/gr-audio/swig/audio_swig_doc_swig_docs/Doxyfile
warning: ignoring unsupported tag `EXTRACT_PRIV_VIRTUAL   =' at line 472, file /cache/workdir/Debian_10_64_0/build_Debian_10_64/build/gr-audio/swig/audio_swig_doc_swig_docs/Doxyfile
warning: ignoring unsupported tag `CLANG_DATABASE_PATH    =' at line 1086, file /cache/workdir/Debian_10_64_0/build_Debian_10_64/build/gr-audio/swig/audio_swig_doc_swig_docs/Doxyfile
warning: ignoring unsupported tag `HTML_DYNAMIC_MENUS     =' at line 1251, file /cache/workdir/Debian_10_64_0/build_Debian_10_64/build/gr-audio/swig/audio_swig_doc_swig_docs/Doxyfile
warning: ignoring unsupported tag `LATEX_MAKEINDEX_CMD    =' at line 1716, file /cache/workdir/Debian_10_64_0/build_Debian_10_64/build/gr-audio/swig/audio_swig_doc_swig_docs/Doxyfile
warning: ignoring unsupported tag `LATEX_EMOJI_DIRECTORY  =' at line 1858, file /cache/workdir/Debian_10_64_0/build_Debian_10_64/build/gr-audio/swig/audio_swig_doc_swig_docs/Doxyfile
warning: ignoring unsupported tag `XML_NS_MEMB_FILE_SCOPE =' at line 2002, file /cache/workdir/Debian_10_64_0/build_Debian_10_64/build/gr-audio/swig/audio_swig_doc_swig_docs/Doxyfile
```

If those tags aren't actually being used in the GR doxygen docs build, then I'd say to comment then out so-as to remove these warnings.

**Comment by ValZapod** [comment]
Well, bad. Will fix it now.

**Comment by ValZapod** [comment]
So I just manually deleted ONLY those 2 variables in two files and change to 500 that variable /sigh

File: gr-utils/python/modtool/templates/gr-newmod/docs/doxygen/other/group_defs.dox [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*!
- * \defgroup block GNU Radio HOWTO C++ Signal Processing Blocks
+ * \defgroup block GNU Radio C++ Signal Processing Blocks
```
Comment (mbr0wn) [inline gr-utils/python/modtool/templates/gr-newmod/docs/doxygen/other/group_defs.dox]: The HOWTO gets transfomed into OOT name by modtool. The correct fix is to ignore gr-newmod/ entirely; it should not be in the manual.

Comment (ValZapod) [reply gr-utils/python/modtool/templates/gr-newmod/docs/doxygen/other/group_defs.dox]: For that you need to add 
```
@abs_top_srcdir@/gr-utils/python/modtool/templates/gr-newmod/docs/doxygen \ 
@abs_top_builddir@/gr-utils/python/modtool/templates/gr-newmod/docs/doxygen \
```
here, right:
https://github.com/gnuradio/gnuradio/blob/2003dda0593cc000af186532e90e2b77aa4d383e/docs/doxygen/Doxyfile.in#L729-L730

Comment (mbr0wn) [reply gr-utils/python/modtool/templates/gr-newmod/docs/doxygen/other/group_defs.dox]: That looks right.

**Comment by ValZapod** [comment]
You can pull.

ср, 22 янв. 2020 г., 7:57 Martin Braun <notifications@github.com>:

**Pull request merged**